### PR TITLE
Adapt test case to new `MessageFormat` check

### PR DESCRIPTION
**Pull request by wmdietl**

### CODE REVIEW

**Comment by wmdietl** [comment]
Maybe macos-14 is okay and the CI failure is from the update of macos-latest to macos-15: `https://github.com/github/roadmap/issues/986`.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -163,7 +163,7 @@ jobs:
       contents: read
     strategy:
       matrix:
-        os: ['windows-latest', 'macos-latest']
+        os: ['windows-latest', 'macos-13']
```
Comment (wmdietl) [inline .github/workflows/ci.yml:166]:
```suggestion
        os: ['windows-latest', 'macos-14']
```

**Comment by wmdietl** [comment]
@Ao-senXiong Could you try whether the `cftests-junit` fail for you locally?
Around a week ago the MacOS and Windows tests started failing, but the Linux tests still pass.
I don't see what dependency could have caused this.

**Comment by Ao-senXiong** [comment]
> @Ao-senXiong Could you try whether the `cftests-junit` fail for you locally? Around a week ago the MacOS and Windows tests started failing, but the Linux tests still pass. I don't see what dependency could have caused this.

Sorry I did not see the comment earlier. I will try it right away.

**Comment by Ao-senXiong** [comment]
> @Ao-senXiong Could you try whether the `cftests-junit` fail for you locally? Around a week ago the MacOS and Windows tests started failing, but the Linux tests still pass. I don't see what dependency could have caused this.

@wmdietl I just tried and all test suite passed locally.

**Comment by wmdietl** [comment]
@Ao-senXiong Thanks for checking. It's very odd...
Both Windows and MacOS start failing with this test:

````
I18nFormatterTest > run[/Users/runner/work/checker-framework/checker-framework/checker/tests/i18n-formatter] FAILED
    java.lang.AssertionError: 69 out of 70 expected diagnostics were found.
    1 unexpected diagnostic was found:
      I18nFormat.java:18: error: (i18nformat.string.invalid)
    1 expected diagnostic was not found:
      I18nFormat.java:18: warning: (i18nformat.missing.arguments)
````

I don't see what change in dependency would suddenly cause this breakage. Do you have any idea?

**Comment by Ao-senXiong** [comment]
> @Ao-senXiong Thanks for checking. It's very odd... Both Windows and MacOS start failing with this test:
> 
> ```
> I18nFormatterTest > run[/Users/runner/work/checker-framework/checker-framework/checker/tests/i18n-formatter] FAILED
>     java.lang.AssertionError: 69 out of 70 expected diagnostics were found.
>     1 unexpected diagnostic was found:
>       I18nFormat.java:18: error: (i18nformat.string.invalid)
>     1 expected diagnostic was not found:
>       I18nFormat.java:18: warning: (i18nformat.missing.arguments)
> ```
> 
> I don't see what change in dependency would suddenly cause this breakage. Do you have any idea?

Odd. I tried the test case and locally it works fine with correct warning message. Let me try to use the CI environment and see whether I can reproduce this issue.
```
checker/tests/i18n-formatter/I18nFormat.java:18: warning: [i18nformat.missing.arguments] missing arguments; expected 44445 but 1 given
        MessageFormat.format("''{0, time, short}''{1}{2, time} {33, number}{44444}'{''''", 0);
```

**Comment by wmdietl** [comment]
The `44444 exceeds the ArgumentIndex implementation limit` error was introduced by https://github.com/openjdk/jdk/commit/d9f638303737c4e5174e5cf21ad2f634cc0b1494
It looks like the MacOS and Windows builds already contain a version with the [backport to 21](https://github.com/openjdk/jdk21u/commit/2058719a38185424753f01e6121fe097c42b55e4).
So eventually, this would have also started failing on the other OSs.

Using a value that is within the range is fine for the test case.

**Comment by wmdietl** [comment]
I'll ignore the JDK 24 failure in Error Prone and hope that that get's fixed by #956.
No idea why the `Dependency review` sometimes fails.

**Pull request merged**